libsolidity/formal/SMTEncoder.cpp

```diff
@@ -3174,10 +3174,21 @@ vector<smtutil::Expression> SMTEncoder::symbolicArguments(FunctionCall const& _f
 		firstParam = 1;
 	}
 
-	solAssert((arguments.size() + firstParam) == functionParams.size(), "");
-	for (unsigned i = 0; i < arguments.size(); ++i)
-		args.push_back(expr(*arguments.at(i), functionParams.at(i + firstParam)->type()));
-
+	if (_funCall.isSuffixCall() && functionParams.size() == 2)
+	{
+		auto literalType = dynamic_cast<RationalNumberType const*>(arguments[0]->annotation().type);
+		solAssert(literalType);
+		auto&& [mantissa, exponent] = literalType->fractionalDecomposition();
+		solAssert(mantissa && exponent);
+		args.push_back(smtutil::Expression(mantissa->literalValue(nullptr)));
+		args.push_back(smtutil::Expression(exponent->literalValue(nullptr)));
```

Comment:
Looks like negative literals require special handling (as 2's complement via `u2s()`) so you may also want to assert that mantissan and exponent are not negative (with `RationalNumberLiteral::isNegative()`).